atproto/identity/handle.go

```diff
@@ -0,0 +1,101 @@
+package identity
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/bluesky-social/indigo/atproto/syntax"
+)
+
+// Does not cross-verify, just does the handle resolution step.
+func ResolveHandleDNS(ctx context.Context, handle syntax.Handle) (syntax.DID, error) {
+	// TODO: timeout
+	// TODO: mechanism to control NDS better; context? separate method?
+
+	res, err := net.LookupTXT("_atproto." + handle.String())
+	// look for NXDOMAIN
+	var dnsErr *net.DNSError
+	if errors.As(err, &dnsErr) {
+		if dnsErr.IsNotFound {
+			return "", ErrHandleNotFound
+		}
+	}
+	if err != nil {
+		return "", fmt.Errorf("handle DNS resolution failed: %w", err)
+	}
+
+	for _, s := range res {
+		if strings.HasPrefix(s, "did=") {
+			parts := strings.SplitN(s, "=", 2)
+			did, err := syntax.ParseDID(parts[1])
+			if err != nil {
+				return "", fmt.Errorf("invalid DID in handle DNS record: %w", err)
+			}
+			return did, nil
+		}
+	}
+	return "", ErrHandleNotFound
+}
+
+func ResolveHandleWellKnown(ctx context.Context, handle syntax.Handle) (syntax.DID, error) {
+	// TODO: timeout
+	// TODO: could pull a client or transport from context?
+	c := http.DefaultClient
```

Comment:
I (try to) never use DefaultClient because there's no telling who else might be using it